From b70554321ae2536b86c92139c997d8dea7097474 Mon Sep 17 00:00:00 2001
From: lyintel <yang.a.lu@intel.com>
Date: Thu, 26 May 2022 05:13:36 +0000
Subject: [PATCH] INTERNAL: egl/android: Add 3 more video format for Android

This change is to help add NV12, NV12_Y_TILED_INTEL and P010
formats for Android.

TESTS: The following CtsViewTestCases cases could pass
android.view.cts.PixelCopyTest#testVideoProducer
android.view.cts.SurfaceViewSyncTest#testVideoSurfaceViewCornerCoverage
android.view.cts.SurfaceViewSyncTest#testVideoSurfaceViewTranslate
android.view.cts.SurfaceViewSyncTest#testVideoSurfaceViewRotated
android.view.cts.SurfaceViewSyncTest#testVideoSurfaceViewEdgeCoverage

Signed-off-by: Lin Johnson <johnson.lin@intel.com>
Signed-off-by: Kevin Strasser <kevin.strasser@intel.com>
Signed-off-by: Chenglei Ren <chenglei.ren@intel.com>
---
 src/egl/drivers/dri2/platform_android.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/egl/drivers/dri2/platform_android.c b/src/egl/drivers/dri2/platform_android.c
index e47bb6125fb..41efa05cf63 100644
--- a/src/egl/drivers/dri2/platform_android.c
+++ b/src/egl/drivers/dri2/platform_android.c
@@ -69,6 +69,15 @@ struct droid_yuv_format {
    int fourcc; /* DRM_FORMAT_ */
 };
 
+/* This enumeration can be deleted if Android defined it in
+ * system/core/include/system/graphics.h
+ */
+enum {
+   HAL_PIXEL_FORMAT_NV12_Y_TILED_INTEL = 0x100,
+   HAL_PIXEL_FORMAT_NV12 = 0x10F,
+   HAL_PIXEL_FORMAT_P010_INTEL = 0x110
+};
+
 /* The following table is used to look up a DRI image FourCC based
  * on native format and information contained in android_ycbcr struct. */
 static const struct droid_yuv_format droid_yuv_formats[] = {
@@ -77,6 +86,9 @@ static const struct droid_yuv_format droid_yuv_formats[] = {
    { HAL_PIXEL_FORMAT_YCbCr_420_888, YCbCr, 1, DRM_FORMAT_YUV420 },
    { HAL_PIXEL_FORMAT_YCbCr_420_888, YCrCb, 1, DRM_FORMAT_YVU420 },
    { HAL_PIXEL_FORMAT_YV12,          YCrCb, 1, DRM_FORMAT_YVU420 },
+   { HAL_PIXEL_FORMAT_NV12,            YCbCr, 2, DRM_FORMAT_NV12 },
+   { HAL_PIXEL_FORMAT_NV12_Y_TILED_INTEL, YCbCr, 2, DRM_FORMAT_NV12 },
+   { HAL_PIXEL_FORMAT_P010_INTEL,      YCbCr, 4, DRM_FORMAT_P010 },
    /* HACK: See droid_create_image_from_prime_fds() and
     * https://issuetracker.google.com/32077885. */
    { HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED, YCbCr, 2, DRM_FORMAT_NV12 },
-- 
2.40.1

